src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.WorkerThread.cs

```diff
@@ -71,6 +71,31 @@ private static void WorkerThreadStart()
                             CultureInfo.CurrentCulture = CultureInfo.InstalledUICulture;
                             CultureInfo.CurrentUICulture = CultureInfo.InstalledUICulture;
                         }
+                        else
+                        {
+                            // If we woke up but couldn't find a request, we need to update s_separated.counts that we are done working for now
```

Comment:
Would be good to refactor this code into a separate function as it seems to be duplicated from the path where Dispatch returns true, maybe something like OnNoWorkAvailable